src/__internal__/input/input-presentation.style.ts

```diff
@@ -114,7 +115,10 @@ const InputPresentationStyle = styled.div<
       border-color: var(--colorsUtilityReadOnly600);
     `}
 
-  ${({ align }) => align === "right" && "flex-direction: row-reverse"}
+  ${({ align, prefix }) =>
+    align === "right" &&
+    `flex-direction: ${prefix ? "row" : "row-reverse"};
+  `}
```

Comment:
**suggestion(non-blocking):** What do you think about simplifying this like so?
```suggestion

```
As the `InputPresentationStyle` is the container of the `<input>` and prefix, `flex-direction="row-reverse"` doesn't control the `<input>`'s text alignment. 

This way we can decouple the CSS of the `prefix` and `align` props, which will help with maintenance.

Reply:
Hi mate, agreed 👍, was considering just removing the whole thing entirely. After doing a bit of digging, the original code was introduced in [this PR](https://github.com/Sage/carbon/commit/0015029c46d75f68d60884019d59e5d01afd2ce4) as part of input validations, so it may serve some purpose. If we remove and no tests fail, and no chromatic regressions are flagged, I'm happy to do so. Will wait and see what a second reviewer thinks as well :)